presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSegmentPageSource.java

```diff
@@ -295,6 +300,66 @@ else if (javaType.equals(Slice.class)) {
         }
     }
 
+    private void writeArrayBlock(BlockBuilder blockBuilder, Type columnType, int columnIndex)
+    {
+        for (int rowIndex = 0; rowIndex < currentDataTable.getDataTable().getNumberOfRows(); rowIndex++) {
+            ColumnDataType columnPinotType = currentDataTable.getDataTable().getDataSchema().getColumnDataType(columnIndex);
+            Type columnPrestoType = ((ArrayType) columnType).getElementType();
+            BlockBuilder childBuilder = blockBuilder.beginBlockEntry();
+            switch (columnPinotType) {
+                case INT_ARRAY:
+                    int[] intArray = currentDataTable.getDataTable().getIntArray(rowIndex, columnIndex);
+                    for (int i = 0; i < intArray.length; i++) {
+                        columnPrestoType.writeLong(childBuilder, intArray[i]);
```

Comment:
Just confirming: This should be writeLong instead of writeInt ? Perhaps add a comment as to why.

Reply:
both the numeric types handle the `writeLong` method and handle it accordingly. So whatever the type instance will handle or throw exception accordingly. Adding a comment (missed in this iteration)

Reply:
Example: https://github.com/prestodb/presto/blob/master/presto-common/src/main/java/com/facebook/presto/common/type/IntegerType.java#L45